fog/view/server/src/sharding_strategy.rs

```diff
@@ -0,0 +1,274 @@
+    fn should_process_block(&self, block_index: BlockIndex) -> bool;
+
+    /// Returns true if the Fog View Store is ready to serve TxOuts to the
+    /// client.
+    ///
+    /// Different sharding strategies might be ready to serve TxOuts when
+    /// different conditions have been met.
+    fn is_ready_to_serve_tx_outs(&self, processed_block_count: BlockCount) -> bool;
+
+    /// Returns the block range that this sharding strategy is responsible for.
+    fn get_block_range(&self) -> BlockRange;
+}
+
+/// Determines whether or not to process a block's TxOuts based on the "epoch"
+/// sharding strategy, in which a block is processed IFF it falls within the
+/// contiguous range of blocks.
+///
+/// In practice, the set of Fog View Shards will contain overlapping
+/// [epoch_block_ranges] in order to obfuscate which shard processed the TxOuts.
+#[derive(Clone, Serialize)]
+pub struct EpochShardingStrategy {
+    /// If a block falls within this range, then the Fog View Store should
+    /// process its TxOuts.
+    epoch_block_range: BlockRange,
+}
+
+impl TryFrom<FogViewStoreUri> for EpochShardingStrategy {
+    type Error = String;
+
+    fn try_from(src: FogViewStoreUri) -> Result<Self, Self::Error> {
+        let sharding_strategy_string = src
+            .get_param("sharding_strategy")
+            .unwrap_or_else(|| "default".to_string());
+        EpochShardingStrategy::from_str(&sharding_strategy_string)
+    }
+}
+
+impl ShardingStrategy for EpochShardingStrategy {
+    fn should_process_block(&self, block_index: BlockIndex) -> bool {
+        self.epoch_block_range.contains(block_index)
+    }
+
+    fn is_ready_to_serve_tx_outs(&self, processed_block_count: BlockCount) -> bool {
+        self.have_enough_blocks_been_processed(processed_block_count)
+    }
+
+    fn get_block_range(&self) -> BlockRange {
+        self.epoch_block_range.clone()
+    }
+}
+
+impl Default for EpochShardingStrategy {
+    fn default() -> Self {
+        Self {
+            epoch_block_range: BlockRange::new(0, u64::MAX),
+        }
+    }
+}
+
+impl ToString for EpochShardingStrategy {
+    fn to_string(&self) -> String {
+        let start_block = self.epoch_block_range.start_block;
+        let end_block = self.epoch_block_range.end_block;
+        format!("{start_block}{BLOCK_RANGE_DELIMITER}{end_block}")
+    }
+}
+
+impl EpochShardingStrategy {
+    #[allow(dead_code)]
+    pub fn new(epoch_block_range: BlockRange) -> Self {
+        Self { epoch_block_range }
+    }
+
+    fn have_enough_blocks_been_processed(&self, processed_block_count: BlockCount) -> bool {
+        if self.is_first_epoch() {
+            return true;
+        }
+
+        let epoch_block_range_length =
+            self.epoch_block_range.end_block - self.epoch_block_range.start_block;
+        let minimum_processed_block_count = epoch_block_range_length / 2;
+
+        u64::from(processed_block_count) >= minimum_processed_block_count
```

Comment:
:question: I'm probably missing something but won't this prevent the last epoch from serving blocks until it's half full
[0 - 10][10 - 20][30 - 40]
If the block height is only 33, would one not be able to get served blocks 31 and 32? since 30-40 is no thalf full?